vault/resource_database_secrets_mount.go

```diff
@@ -0,0 +1,334 @@
+				Type: schema.TypeString,
+			},
+		},
+		"data": {
+			Type:        schema.TypeMap,
+			Optional:    true,
+			Description: "A map of sensitive data to pass to the endpoint. Useful for templated connection strings.",
+			// TODO: revert to true
+			Sensitive: false,
+		},
+	}
+}
+
+func setCommonDatabaseSchema(s schemaMap) schemaMap {
+	for k, v := range getCommonDatabaseSchema() {
+		s[k] = v
+	}
+	return s
+}
+
+func databaseSecretsMountCreateOrUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	var root string
+	if d.IsNewResource() {
+		root = d.Get("path").(string)
+		if err := createMount(d, client, root, "database"); err != nil {
+			return err
+		}
+	} else {
+		if err := mountUpdate(d, meta); err != nil {
+			return err
+		}
+		root = d.Id()
+	}
+	d.SetId(root)
+
+	var count int
+	seen := make(map[string]bool)
+	for _, engine := range dbEngines {
+		if v, ok := d.GetOk(engine.Name()); ok {
+			for i := range v.([]interface{}) {
+				prefix := engine.ResourcePrefix(i)
+				name := d.Get(prefix + "name").(string)
+				path := databaseSecretBackendConnectionPath(root, name)
+				if _, ok := seen[name]; ok {
+					return fmt.Errorf("duplicate name %q for engine %#v", name, engine)
+				}
+				seen[name] = true
+				if err := writeDatabaseSecretConfig(d, client, engine, i, true, path); err != nil {
+					return err
+				}
+				count++
+			}
+		}
+	}
+
+	if err := databaseSecretsMountRead(d, meta); err != nil {
+		return err
+	}
+
+	action := "Created"
+	if !d.IsNewResource() {
+		action = "Updated"
+	}
+
+	log.Printf("[DEBUG] %s %d database connections under %q", action, count, root)
+	return nil
+}
+
+func databaseSecretsMountRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	if err := readMount(d, meta, true); err != nil {
+		return err
+	}
+
+	root := d.Id()
+	if root == "" {
+		return nil
+	}
+
+	resp, err := client.Logical().List(root + "/config")
+	if err != nil {
+		return err
+	}
+
+	if resp == nil {
+		return nil
+	}
+
+	store := &dbConfigStore{}
+	if v, ok := resp.Data["keys"]; ok {
+		for _, v := range v.([]interface{}) {
+			if err := readDBEngineConfig(d, client, store, v.(string)); err != nil {
+				return err
+			}
+		}
+
+		for k, v := range store.Result() {
+			if err := d.Set(k, v); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func databaseSecretsMountDelete(d *schema.ResourceData, meta interface{}) error {
+	return mountDelete(d, meta)
+}
+
+func readDBEngineConfig(d *schema.ResourceData, client *api.Client, store *dbConfigStore, name string) error {
+	root := d.Id()
+
+	path := databaseSecretBackendConnectionPath(root, name)
+	log.Printf("[DEBUG] Reading database connection config %q", path)
+	resp, err := client.Logical().Read(path)
+	if err != nil {
+		return fmt.Errorf("error reading database connection config %q: %s", path, err)
+	}
+	log.Printf("[DEBUG] Read database connection config %q", path)
```

Comment:
nit: per a previous conversation of ours, could we make this `Successfully read...` for better readability and consistency with other debug strings?